extractor/filesystem/os/vscode/extensions/extensions.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package extensions extracts vscode extensions.
+package extensions
```

Comment:
Could put into a new "misc" dir instead of OS
Also I don't think we'll have other vscode related extraction logic so it should be fine to put under a single dir
i.e. something like `filesystem/misc/vscode-extensions`